custom-services/custom-service.md

```diff
@@ -0,0 +1,57 @@
+While it would be something a developer familiar with OpenShift/K8s could do, we can provide a better user experience.
+Currently we allow configuration to be exposed for the mobile enabled services such as Keycloak that we have created specific APBs for, but most real world mobile client applications require some level of custom business logic. It is important that we allow this business logic to be exposed by the mobile developer to their mobile client applications in a consistent intuitive and simple fashion. If possible once the custom service has been created it should be viewed in the UI as just another mobile service.
+
+## Use cases we hope to solve
+- As a mobile developer, I want to expose a service I have already provisioned from the catalog to my mobile clients, so that I can make use of this service in a consistent and simple fashion.
+
+- As a mobile developer, there is a service with custom logic running somewhere that I need to make use of from within my client, I want to easily make this service accessible to my client devices with the needed configuration so that I can consume the custom business logic within my mobile client application.
+
+# Proposed Solution
+
+### Add enable mobile button / link
+To allow a developer to expose a service already provisioned from the service catalog, we could modify the UI view in OpenShift to have a call
+to action (something like) "enable mobile" that would show up on the overview screen inside the ServiceInstance box if mobile was enabled within the cluster and there was at least one mobile client in the namespace. When clicked they would be presented with a modal form that would allow them to specify needed configuration details (url, API Keys, any other key value pairs needed in order to consume the service). The key value pairs would be dynamic (as in they could add as many as needed) there is already options for this type of field in the OpenShift UI (used to specify env vars).
+When saved we would create the required objects (ie the configmap and a secret if needed) in the right format and with the correct labels to allow it to be consumed from the mobile CLI.
+
+**Considerations**
+
+Do we add this button to all services, it doesn't make sense to add it to something like MongoDB for example.
+
+## Create Custom Service Connector APB
+
+To allow mobile developers to expose existing custom logic contained within running services that may or may not be running on an OpenShift cluster, I propose we look at creating a custom service connector APB.
+
+All of our mobile enabled services are provisioned via the catalog. I believe we should follow the same pattern for allowing mobile developers to expose custom services to the mobile application clients. This would give us the following advantages:
+
+- It would create a ServiceInstance that we can then leverage as part of the UI work, ensuring we have a single Object to deal with for all service interactions in the UI and CLI.
```

Comment:
How would that look/feel if there was potentially 2 ServiceInstances for a Service e.g. one for the actual service & one for the Mobile service instance?

Reply:
in the same namespace? If there is an existing service instance then they would be able to click "enable mobile" on that service instance. There shouldn't be a situation from this proposal where two service instances are created

Reply:
ah, misunderstanding. This is fine